eng/pipelines/templates/steps/analyze.yml

```diff
@@ -122,114 +147,194 @@ steps:
       targetType: filePath
       pwsh: true
       filePath: eng/scripts/Invoke-DependencyCheck.ps1
-      arguments: 'sdk/${{ parameters.ServiceDirectory }}'
+      arguments: "$(Packages)"
 
   - script: |
       curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${{parameters.LintVersion}}
       golangci-lint --version
     displayName: 'Install GoLintCLI and dependencies.'
+    condition: succeededOrFailed()
     workingDirectory: $(System.DefaultWorkingDirectory)
     retryCountOnTaskFailure: 3
 
   - pwsh: |
-      $modDirs = ./eng/scripts/get_module_dirs.ps1 '${{ parameters.ServiceDirectory }}'
-      foreach ($md in $modDirs) {
-        pushd $md
-        Write-Host "##[command]Executing golangci-lint run -c $(System.DefaultWorkingDirectory)/eng/.golangci.yml in $md"
-        golangci-lint run -c $(System.DefaultWorkingDirectory)/eng/.golangci.yml
+      $changedServices = "$(ChangedServices)" -split ","
+      $failed = $false
+
+      foreach($serviceDirectory in $changedServices) {
+        $modDirs = ./eng/scripts/get_module_dirs.ps1 "$serviceDirectory"
+        foreach ($md in $modDirs) {
+          pushd $md
+          Write-Host "##[command]Executing golangci-lint run -c $(System.DefaultWorkingDirectory)/eng/.golangci.yml in $md"
+          golangci-lint run -c $(System.DefaultWorkingDirectory)/eng/.golangci.yml
+
+          if ($LASTEXITCODE) {
+            Write-Host "golangci-lint for $serviceDirectory failed. Check above output for details."
+            $failed = $true
+          }
+        }
+      }
+
+      if ($failed) {
+        Write-Host "golangci-lint failed. Check above output for details."
+        exit 1
       }
     displayName: 'Lint'
-    failOnStderr: false
```

Comment:
Did you intend to remove this failOnStderr?

Reply:
I did not!